### PR TITLE
(Add Payments) Added Payment form, API and Payment

### PR DESCRIPTION
**Pull request by aagungm**
Payments form:
- created form for submitting 
  
  API (Api.js) :
- created addPayment()

Payments :
- created function for calling the payment form

### CODE REVIEW

File: scripts/components/PaymentForm.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      <div style={style.formContainer}>
+        <h2 style={style.heading}>Make a Payment </h2>
+        <Paper zDepth={1}>
+          <form style={style.form} onSubmit={this.onSubmit}>
+            <DatePicker
+              name="Date"
+              ref="DatePicker"
+              onChange={this._handleChange}
+              floatingLabelText="Date"/>
+            <TextField
```
Comment (nirrek) [inline scripts/components/PaymentForm.jsx:84]: It doesn't make sense for the tenant to input the property name. The tenant rent's a single property. We know which property this is so we should be doing it automatically. In addition, the tenant's account only ever displays information for a single property. From a user perspective they would go `why the hell do I have to enter the property??? they know what property I rent...`

File: scripts/components/PaymentForm.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        property: this.state.property,
+        amount: this.state.amount
+      },
+      callback: (err, response) => {
+        if (err) {
+          return;
+        }
+        // Clear the form
+        this.setState({
+          date: "",
```
Comment (nirrek) [inline scripts/components/PaymentForm.jsx]: To maintain consistent styling with the codebase, when writing strings in JavaScript we want to use single quotes, not double quotes.

``` javascript
date: '',
property: '',
```

Double quotes are fine for props/attributes in JSX / HTML

**Comment by nirrek** [comment]
@aagungm This feature won't currently work without setting up a server side handler for your POST request. Are you working towards that as well?

**Comment by nirrek** [comment]
Also, before merging this can you also make the payment form hidden by default, with just a button `Make Payment` that sits at the bottom of the table. When the user clicks this button the form will be shown.

**Comment by aagungm** [comment]
@nirrek Ok I will work on those changes.

**Comment by nirrek** [comment]
@aagungm 
Is there a reason you have changed file permissions on a bunch of files?

image

File: scripts/components/PaymentForm.jsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        // ccv: this.state.ccv,
+        amount: this.state.amount
+      },
+      callback: (err, response) => {
+        if (err) {
+          return;
+        }
+        // Clear the form
+        this.setState({
+          fullName: "",
```
Comment (nirrek) [inline scripts/components/PaymentForm.jsx]: Can you follow the style of the codebase. Strings in JavaScript should be single quoted. Eg. `fullName: ''`

File: scripts/components/PaymentForm.jsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    var errorMessage;
+    return (
+      <div style={style.formContainer}>
+        <h2 style={style.heading}>Make a Payment </h2>
+        <Paper zDepth={1}>
+          <form style={style.form} onSubmit={this.onSubmit}>
+            <div style={style.error}> { errorMessage } </div>
+              <TextField
+              value={fullName}
+              name="Full Name on Credit Card"
```
Comment (nirrek) [inline scripts/components/PaymentForm.jsx]: Could you explain the purpose of the `name` attribute used in all the TextField components?

File: scripts/components/PaymentForm.jsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+              value={amount}
+              name="Amount"
+              onChange={this.onChange.bind(this, 'amount')}
+              floatingLabelText="Amount" />
+
+            <RaisedButton
+              type="submit"
+              label="Pay"
+              primary={true}
+              backgroundColor="#2ECC71"
```
Comment (nirrek) [inline scripts/components/PaymentForm.jsx]: Specifying the backgroundColor is no longer required as I have changed the palette for components, so primary={true} is sufficient. Before I updated the palette the primary color was hot pink.

File: scripts/components/Payments.jsx [hunk 6]
```diff
@@ -24,10 +32,23 @@ var Payments = React.createClass({
     })
   },
 
+  onButtonClick(){
+      this.setState({
+          isFormDisplayed: true
+      });
+  },
+
+   componentWillMount() {
```
Comment (nirrek) [inline scripts/components/Payments.jsx:39]: Indentation is all screwed here

**Comment by aagungm** [comment]
@nirrek It because I updated from master but I don't know why the file permissions have changed.

**Pull request merged**